script/tool/test/format_command_test.dart

```diff
@@ -348,4 +494,95 @@ void main() {
           contains('Unable to determine diff.'),
         ]));
   });
+
+  test('Batches moderately long file lists on Windows', () async {
+    mockPlatform.isWindows = true;
+
+    const String pluginName = 'a_plugin';
+    // -1 since the command itself takes some length.
+    const int batchSize = (windowsCommandLineMax ~/ 100) - 1;
+
+    // Make the file list one file longer than would fit in the batch.
+    final List<String> batch1 =
+        _get99CharacterPathExtraFiles(pluginName, batchSize + 1);
+    final String extraFile = batch1.removeLast();
+
+    createFakePlugin(
+      pluginName,
+      packagesDir,
+      extraFiles: <String>[...batch1, extraFile],
+    );
+
+    await runCapturingPrint(runner, <String>['format']);
+
+    // Ensure that it was batched...
+    expect(processRunner.recordedCalls.length, 2);
+    // ... and that the spillover into the second batch was only one file.
+    expect(
+        processRunner.recordedCalls,
+        contains(
+          ProcessCall(
+              getFlutterCommand(mockPlatform),
+              <String>[
+                'format',
+                '$pluginName\\$extraFile',
+              ],
+              packagesDir.path),
+        ));
+  });
+
+  test('Does not batch moderately long file lists on non-Windows', () async {
+    const String pluginName = 'a_plugin';
+    // -1 since the command itself takes some length.
+    const int batchSize = (windowsCommandLineMax ~/ 100) - 1;
+
+    // Make the file list one file longer than would fit in a Windows batch.
+    final List<String> batch =
+        _get99CharacterPathExtraFiles(pluginName, batchSize + 1);
+
+    createFakePlugin(
+      pluginName,
+      packagesDir,
+      extraFiles: batch,
+    );
+
+    await runCapturingPrint(runner, <String>['format']);
+
+    expect(processRunner.recordedCalls.length, 1);
+  });
+
+  test('Batches extremely long file lists on non-Windows', () async {
+    const String pluginName = 'a_plugin';
+    const int commandLineMax = 1000000;
```

Comment:
Unused var `commandLineMax`?

Reply:
Fixed; cruft from before I made the values public for testing.